### PR TITLE
akka.build.version property, #26958

### PR DESCRIPTION
**Pull request by patriknw**
## Purpose

* to support publishing snapshots with same version for Scala 2.12 and 2.13
  with +publish

## References

References #26958

## Changes

* and use timestamped snapshot version if "timestamp" is passed in
* remove stampVersion plugin, since it's no longer needed

## Background Context

So that for example Play can build for both 2.12 and 2.13 with the same Akka snapshot version.

### CODE REVIEW

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -21,14 +24,25 @@ object AkkaBuild {
 
   lazy val buildSettings = Dependencies.Versions ++ Seq(
     organization := "com.typesafe.akka",
-    // use the same value as in the build scope, so it can be overriden by stampVersion
+    // use the same value as in the build scope
     version := (version in ThisBuild).value)
 
+  lazy val currentDateTime = DateTimeFormatter
+    .ofPattern("yyyyMMdd-HHmmss")
+    .format(ZonedDateTime.now(ZoneOffset.UTC))
```
Comment (dwijnand) [inline project/AkkaBuild.scala]: Taking from my comment in https://github.com/akka/akka/issues/26958#issuecomment-493454817, I think a less fragile implementation would be:

```scala
  val timestamp = sys.props.getOrElseUpdate(
    "akka.build.timestamp",
    java.time.format.DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss").format(
      java.time.ZonedDateTime.now(java.time.ZoneOffset.UTC)
    )
  )
```

It also benefits from allowing it to be defined externally.

Comment (patriknw) [reply project/AkkaBuild.scala]: added in 30c0b5e

**Comment by patriknw** [comment]
@dwijnand I have published version `2.6-20190517-130930` if you want to take it for a spin in Play. For 2.12 and 2.13.0-M5, but maybe you are more interested in the 2.5 version. I'll backport this.

**Comment by dwijnand** [comment]
Thanks!  I've created https://github.com/playframework/play-meta/issues/79 to track as a follow-up task (before my weekend GC).

**Comment by patriknw** [comment]
Here is one for Akka 2.5 as well: 2.5-20190517-162558
For Scala 2.11, 2.12 and 2.13.0-M5

**Review by raboof — APPROVED** [review]
Eventually #26675 seems like a nicer solution but until that time this is :+1:

**Pull request merged**